sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -800,7 +800,7 @@ private void openNewChannelWithInjectedDelay(final Promise<Channel> promise, Dur
         // Reduce the connection timeout based on the injected delay
         // The higher delay being injected, then less time left to open connection, then it is easier to get connectionTimeout exception.
         // But we would not want to use a <0 for connectionTimeout setting as netty throw exceptions.
-        long effectiveConnectTimeoutInMillis = Math.max(this.connectTimeoutInMillis - delayInMillis, 10);
+        long effectiveConnectTimeoutInMillis = Math.max(this.connectTimeoutInMillis - delayInMillis, 5);
```

Comment:
why are we changing this?

Reply:
My intention for this PR is to force `ConnectTimeoutException` by injecting high connection establishment delays through fault injection. Internally, to force such a behavior `ChannelOptions#CONNECT_TIMEOUT_MILLIS` was set to 10ms (an artificially low value) to force a timeout through netty. Turns out even with 10ms, connections were being established (not fully sure why?) so I reduced 10ms to 5ms.